### ISSUE TITLE
contains() method returns partial last match

### ISSUE THREAD

**Issue by bahmutov** [cypress-io/cypress#158]
Given HTML fragment
```html
<ul>
  <li>foo</li>
  <li>bar</li>
  <li>bar foo</li>
</ul
```
then `contains` returns *last* element with partial match
```js
cy.get('ul').contains('li', 'foo') 
// returns <li>bar foo</li>
```

It would be nice to prefer exact string match in case of multiple matches.
We also just wrote our own regular expression command for this case
```js
function textCommand (subject, selector, regex) {
  if (!subject || !Cypress.Utils.hasElement(subject)) {
    subject = this.prop('withinSubject') || this.$$('body')
  }

  if (typeof regex === 'string') {
    regex = new RegExp(regex)
  }

  var log = Cypress.Log.command({
    name: 'text',
    message: ['finding text', selector, regex],
    $el: subject,
    onConsole: function () {
      return {
        Subject: subject,
        selector: selector,
        regex: regex
      }
    }
  })

  const items = Array.from(subject.find(selector))
  const found = items.find((el) => regex.test(el.outerText))
  log.snapshot().end()
  return found
}
// use like cy.get('.element').text('span', /foo$/)
Cypress.addDualCommand('text', textCommand)
```

**Comment by brian-mann** [comment]
Doing some preliminary testing, this is actually caused by you passing the `li` selector as the first argument. If you hypothetically only passed in the content without the selector this would work.

**Comment by brian-mann** [comment]
Okay I was able to fix both of your requests, however I don't think we'll opt into preferring "exact string matches" over partial matches.

In `0.16.2` the first `<li>` will be returned instead of the last. I'm not sure why we had code that was returning the last element when the `selector` was provided as an argument. After I removed that code all 70+ tests around `cy.contains` still passed. Must have been an oversight.

Additionally I was able to add in `RegExp` support without any performance problems. There could be some weird edge cases around this but I added a bunch of tests and they all pass.
